File: 백준/Silver/1920. 수 찾기/수 찾기.cc
#include<bits/stdc++.h>

using namespace std;

int N;
int M;
int num[100001];

int main()
{
	ios_base::sync_with_stdio(0);
	cin.tie(0);
	cin>>N;
	for(int i=1;i<=N;i++)
		cin>>num[i];
	sort(num+1,num+1+N);
	cin>>M;
	for(int i=1;i<=M;i++)
	{
		int n;
		cin>>n;
		if(binary_search(num+1,num+1+N,n))
			cout<<"1\n";
		else
			cout<<"0\n";
	}
}

File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.io.*;
import java.util.*;

public class Main {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st;
    int[] num = new int[111_111];
    int n;

    boolean binarySearch(int x) {
        int l = 0, r = n - 1;
        while (l <= r) {
            int mid = (l + r) / 2;
            if (num[mid] == x)
                return true;
            if (num[mid] > x)
                r = mid - 1;
            else
                l = mid + 1;
        }
        return false;
    }

    void solution() throws Exception {
        n = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++)
            num[i] = Integer.parseInt(st.nextToken());
        Arrays.sort(num, 0, n);
        StringBuilder sb = new StringBuilder();
        int m = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < m; i++) {
            int x = Integer.parseInt(st.nextToken());
            if (binarySearch(x))
                sb.append("1\n");
            else
                sb.append("0\n");
        }
        System.out.println(sb);
    }

    public static void main(String[] args) throws Exception {
        new Main().solution();
    }
}